gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java

```diff
@@ -56,20 +65,17 @@
   }
 
   @Override
-  public Map<String, String> getPathParams(ApiMessage apiMessage, Set<String> paramNames) {
-    Map<String, String> pathParams = new HashMap<>();
-    Map<String, List<String>> pathParamMap = apiMessage.populateFieldsInMap(paramNames);
-    Iterator<Map.Entry<String, List<String>>> iterator = pathParamMap.entrySet().iterator();
-    while (iterator.hasNext()) {
-      Map.Entry<String, List<String>> pair = iterator.next();
-      pathParams.put(pair.getKey(), pair.getValue().get(0));
-    }
-    return pathParams;
+  public Map<String, String> getPathParams(T apiMessage, String resourceNameField) {
+    Map<String, List<String>> pathParamMap =
+        apiMessage.populateFieldsInMap(Collections.singleton(resourceNameField));
```

Comment:
I'm concerned about efficiency here - we will be constructing a map with only a single item, for every request (since every request has a resource name field that maps to path params).

Reply:
The `<T extends ApiMessage> T` message type is dumb and doesn't know which of its fields contain the path param. Therefore, the `ApiMessage.populateFieldsInMap()` method is necessary to extract the path param fields, and this method returns the Map.

Reply:
We can assume there will only be one resource field that maps to the path, so we should add a new method in ApiMessage to get a single field out.

Reply:
Done.